server/src/main/java/org/elasticsearch/search/fetch/FetchSubPhase.java

```diff
@@ -39,13 +40,15 @@
         private IndexSearcher searcher;
         private LeafReaderContext readerContext;
         private int docId;
+        private final SourceLookup sourceLookup = new SourceLookup();
```

Comment:
Do we always need to create `SourceLookup` object here?  What about cases where we don't need _source?

Reply:
To me the optimization might not be worth it, it's not an expensive object to construct and we only create one `SourceLookup` per hit?